typed-racket-doc/typed-racket/scribblings/reference/special-forms.scrbl

```diff
@@ -814,5 +804,45 @@ Performs type checking of forms entered at the read-eval-print loop.  The
 @racket[#%top-interaction] form also prints the type of @racket[form] after
 type checking.}
 
+@section{Special Structure Type Properties}
+@defthing[prop:procedure struct-type-property?]{
+Unlike many other structure type properties, @racket[prop:procedure] does not
+have predefined types for its property values. When a structure is assocatied
+with @racket[prop:procedure], its constructors' return type is an intersection
+type of the structure type and a function type specified by the property
+value.
+}
+
+@ex[
+(struct animal ([a : Number] [b : (-> Number Number)])
+  #:property prop:procedure
+  (struct-field-index b))
+(animal 2 add1)
+(struct plant ([a : Number])
+  #:property prop:procedure
+  (lambda ([me : plant] [a1 : String]) : Number
+    (+ (plant-a me) (string-length a1))))
+(plant 31)
+]
+
+When a structure's base structure is procedural, so is the structure.
```

Comment:
I think "is procedural" is a bad phrase -- just spell out what we mean.

Reply:
use "has a prop:procedure" or something